### ISSUE TITLE
ModuleStore : UCD-SNMP-MIB.mib gets loaded but has no providers

### ISSUE THREAD

**Issue by DineshDevaraj** [markabrahams/node-net-snmp#151]
Hi All,

I have shown the code and the output below
As you can see I am using ModuleStore to load UCD-SNMP-MIB.mib
The MIB gets loaded but has no providers
Am I missing any step or is this an expected behavior ?

```
let snmp = require("net-snmp")
let mibStore = snmp.createModuleStore()

mibStore.loadFromFile("/root/keysight/snmp-agent/mib-files/UCD-SNMP-MIB.mib")

let moduleNames = mibStore.getModuleNames()
console.log("Module name : ", moduleNames)

let providers = mibStore.getProvidersForModule(moduleNames[0])
console.log("Providers : ", providers)

```

**#Output**

```
root@c8c1c6a106c2:~/snmp-agent# node load-ucd-mib.js
Module name :  [ 'UCD-SNMP-MIB' ]
Providers :  []
root@c8c1c6a106c2:~/snmp-agent#
```

**Comment by derrell** [comment]
This is a MIB parser bug. The problem is that in UCD-SNMP-MIB.mib, the definition of `Float` is as follows:

```
Float ::= TEXTUAL-CONVENTION
    STATUS      current
    DESCRIPTION
        "A single precision floating-point number.  The semantics
         and encoding are identical for type 'single' defined in
         IEEE Standard for Binary Floating-Point,
         ANSI/IEEE Std 754-1985.
         The value is restricted to the BER serialization of
         the following ASN.1 type:
             FLOATTYPE ::= [120] IMPLICIT FloatType
         (note: the value 120 is the sum of '30'h and '48'h)
         The BER serialization of the length for values of
         this type must use the definite length, short
         encoding form.

         For example, the BER serialization of value 123
         of type FLOATTYPE is '9f780442f60000'h.  (The tag
         is '9f78'h; the length is '04'h; and the value is
         '42f60000'h.) The BER serialization of value
         '9f780442f60000'h of data type Opaque is
         '44079f780442f60000'h. (The tag is '44'h; the length
         is '07'h; and the value is '9f780442f60000'h."
    SYNTAX Opaque (SIZE (7))
```

Note that on the second-to-last line of the description, there is an opening parenthesis but no closing parenthesis on the statement beginning, "(The tag is '44'h", which causes nothing else in the file to be parsed (including the SYNTAX clause which is indicated incorrectly in the resulting entry array. Parenthesis matching should not be required within a quoted description.

As a work-around, add a closing parenthesis just before the closing quote, and I think you should be good to go, for now.

**Comment by DineshDevaraj** [comment]
The work-around works. Thanks!

**Comment by markabrahams** [comment]
Hi @DineshDevaraj - I've fixed the parenthesis parsing bug in master now and published the fix in version 3.0.7 of the npm.

Thanks @derrell for suggesting the interim workaround here too!
